poclu/misc.c

```diff
@@ -673,3 +674,15 @@ poclu_load_program (cl_context context, cl_device_id device,
                                       spirv, poclbin, explicit_binary,
                                       extra_build_opts, p);
 }
+
+cl_half
+poclu_float_to_cl_half (float value)
```

Comment:
This might be problematic. poclu was originally meant to be an OpenCL-implementation independent utility library for tests etc. thus it should not call pocl internal utilities directly as those are not made available in the produced libOpenCL.so/pocl, especially if one links via ICD. If we don't need the halfs in the tests, I think we can remove it from poclu and leave it to pocl_util or even duplicate the code for now via including .c. I cannot think of a better solution quickly.

Reply:
I see I'll have to rebase commit out at some point. But in any case, poclu is built with the pocl_cl_half_util source file, so it does not have a dependency on pocl. And while I was searching for the file, I found a third [cl_half](https://github.com/pocl/pocl/blob/main/include/CL/cl_half.h) file from khronos, maybe that one should replace the misc and internal cl_half conversion functions.